src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -29364,6 +29709,90 @@ static void GetValue2<T>(T x, out T y)
                 );
         }
 
+        [Fact, WorkItem(39922, "https://github.com/dotnet/roslyn/issues/39922")]
+        public void EnforcedInMethodBody_DoesNotReturn()
+        {
+            var source = @"
+#nullable enable
+using System.Diagnostics.CodeAnalysis;
+class C
+{
+    [DoesNotReturn]
+    static void ActuallyReturns()
+    {
+    } // 1
+
+    [DoesNotReturn]
+    static bool ActuallyReturns2()
+    {
+        return true; // 2
+    }
+
+    [DoesNotReturn]
+    static bool ActuallyReturns3(bool b)
+    {
+        if (b)
+            throw null!;
+        else
+            return true; // 3
+    }
+
+    [DoesNotReturn]
+    static bool ActuallyReturns4()
+        => true; // 4
+
+    [DoesNotReturn]
+    static void NeverReturns()
+    {
+        throw null!;
+    }
+
+    [DoesNotReturn]
+    static bool NeverReturns2()
+    {
+        throw null!;
+    }
+
+    [DoesNotReturn]
+    static bool NeverReturns3()
+        => throw null!;
+
+    [DoesNotReturn]
```

Comment:
Is `DoesNotReturn` supported on property/event accessors? Do we have analogous tests for those scenarios?

Reply:
That is not supported. The attribute has `    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]`
`

---
In reply to: [369755491](https://github.com/dotnet/roslyn/pull/41098#discussion_r369755491) [](ancestors = 369755491)